File: 3471-find-the-largest-almost-missing-integer/3471-find-the-largest-almost-missing-integer.py
class Solution:
    def largestInteger(self, nums: List[int], k: int) -> int:
        if k == len(nums):
            return max(nums)
        res = -1
        freqs = {}
        for num in nums:
            freqs[num] = freqs.get(num, 0) + 1
        if k == 1:
            for x, f in freqs.items():
                if f == 1:
                    res = max(res, x)
        else:
            if freqs[nums[0]] == 1:
                res = max(res, nums[0])
            if freqs[nums[-1]] == 1:
                res = max(res, nums[-1])
        return res
        

File: 3471-find-the-largest-almost-missing-integer/3471-find-the-largest-almost-missing-integer.cpp
class Solution {
public:
    int largestInteger(vector<int>& nums, int k) {
        int n = nums.size();
        if(k == n)
        {
            int mx = 0;
            for(int i = 0; i < n; i++)
            {
                mx = max(mx, nums[i]);
            }
            return mx;
        }  
        int res = -1;
        unordered_map<int, int> freqs;
        for(int i = 0; i < n; i++)
        {
            freqs[nums[i]]++;
        }
        if(k == 1)
        {
            for(auto& [x, f] : freqs)
            {
                if(f == 1)
                {
                    res = max(res, x);
                }
            }
        }
        else
        {
            if(freqs[nums[0]] == 1)
            {
                res = max(res, nums[0]);
            }
            if(freqs[nums[n - 1]] == 1)
            {
                res = max(res, nums[n - 1]);
            }
        }
        return res;
    }
};